CompressionCheck.py

```diff
@@ -0,0 +1,63 @@
+# vim:sw=4:et
+#############################################################################
+# File          : CompressionCheck.py
+# Package       : rpmlint
+# Author        : Bernhard M. Wiedemann
+# Purpose       : Check for compressed files using correct extension
+#############################################################################
+
+import AbstractCheck
+import Config
+import Filter
+import os
+import stat
+import string
+
+magic = {
+    "gz": "\x1f\x8b",
+    "tgz": "\x1f\x8b",
+    "bz2": "BZ",
+    "xz": "\xfd\x37"}
+
+
+def get_ext(file):
+    return file.split("/")[-1].split(".")[-1]
+
+
+def get_filestart(file):
+    f = open(file, "r")
+    return f.read(2)
+
+
+def wrong_compression(file):
+    ext = get_ext(file)
+    if ext in magic and magic[ext] != get_filestart(file):
+        return 1
+    return 0
+
+
+class CompressionCheck(AbstractCheck.AbstractCheck):
+    def __init__(self):
+        self.map = []
+        AbstractCheck.AbstractCheck.__init__(self, "CompressionCheck")
+
+    def check(self, pkg):
```

Comment:
mostly, but in principle, binary rpms could also contain .gz etc files, which we might want to check.